src/pybind/mgr/volumes/fs/operations/versions/__init__.py

```diff
@@ -95,7 +95,10 @@ def get_subvolume_object(self, mgr, fs, vol_spec, group, subvolname, upgrade=Tru
             subvolume.discover()
             self.upgrade_to_v2_subvolume(subvolume)
             version = int(subvolume.metadata_mgr.get_global_option('version'))
-            return self._get_subvolume_version(version)(mgr, fs, vol_spec, group, subvolname, legacy=subvolume.legacy_mode)
+            subvolume_version_object = self._get_subvolume_version(version)(mgr, fs, vol_spec, group, subvolname, legacy=subvolume.legacy_mode)
+            subvolume_version_object.metadata_mgr.refresh()
+            subvolume_version_object.clean_stale_snapshot_metadata()
```

Comment:
can it be tested that stale snapshot metadata gets cleaned up?

Reply:
Sure I will add the test case for this.

Reply:
Done